### PR TITLE
refactor: moved each tauri-plugin to own rust-crate

### LINKED ISSUE #141: move each tauri plugin in it's own crate
Create a cargo workspace at the rep-root and create a lib for each tauri plugin.
It may be necessary to modify tauri to look for changes in these crates too.

### PR DESCRIPTION
**Pull request by crapStone**
closes #141

### CODE REVIEW

**Review by Epsilon02 — DISMISSED** [review]

**Review by 6543 — CHANGES_REQUESTED** [review]
noch ned mergen

**Review by 6543 — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,14 +18,9 @@ jobs:
         run: rustup default stable
 
       - name: Rust Linting
-        run: |
-          ls -la /usr/local/bin/
-          cd ./src-tauri
-          cargo clippy
+        run: ./contrib/for-each-cargo-dir clippy
```
Comment (crapStone) [inline .github/workflows/test.yml:21]: where is that file?

Comment (6543) [inline .github/workflows/test.yml:21]:
```suggestion
        run: ./contrib/for-each-cargo clippy
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -18,14 +18,9 @@ jobs:
...
       - name: Rust Tests
-        run: |
-          cd ./src-tauri
-          cargo test
+        run: ./contrib/for-each-cargo-dir test
```
Comment (6543) [inline .github/workflows/test.yml:23]:
```suggestion
        run: ./contrib/for-each-cargo test
```

**Review by 6543 — COMMENTED** [review]
...

File: contrib/for-each-cargo [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+rust_root="$(dirname "$0")/../src-tauri"
```
Comment (6543) [inline contrib/for-each-cargo:3]:
```suggestion
rust_root="$(dirname "$0")/../src-tauri"
echo "rust_root=$rust_root"
```

Comment (6543) [inline contrib/for-each-cargo:3]:
```suggestion
cd "$(dirname "$0")/../src-tauri"
rust_root="$(pwd)"
```

**Review by 6543 — APPROVED** [review]

**Review by Epsilon02 — APPROVED** [review]

**Pull request merged**